test/integration_test.py

```diff
@@ -1035,3 +1043,78 @@ def test_local_filtered_index_endpoints(self):
                                              catalog=self.catalog)).url
         response = requests.get(url)
         self.assertEqual(200, response.status_code)
+
+
+class CanBundleScriptIntegrationTest(AzulTestCase):
+    scripts_root = f'{config.project_root}/scripts'
+
+    def _test_catalog(self, catalog: config.Catalog):
+        with self.subTest(catalog=catalog.name,
+                          repository=catalog.plugins['repository']):
+            fqid = self.bundle_fqid(catalog.name)
+            with tempfile.TemporaryDirectory() as d:
+                self._can_bundle(source=str(fqid.source.spec),
+                                 uuid=fqid.uuid,
+                                 version=fqid.version,
+                                 output_dir=d)
+                parts = ['manifest', 'metadata']
+                expected_files = sorted(f'{fqid.uuid}.{part}.json' for part in parts)
+                generated_files = sorted(os.listdir(d))
+                self.assertListEqual(generated_files, expected_files)
+
+    def test_can_bundle_configured_catalogs(self):
+        for catalog_name, catalog in config.catalogs.items():
+            if catalog.is_integration_test_catalog:
+                self._test_catalog(catalog)
+                break
+
+    canned_repo = 'https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests'
+
+    @mock.patch.dict(os.environ,
+                     azul_canned_sources=f'{canned_repo}:')
+    @mock.patch.object(azul.Config, 'catalogs', new_callable=PropertyMock)
+    # Must be mocked independently due to caching
+    @mock.patch.object(azul.Config, 'default_catalog', new_callable=PropertyMock)
+    def test_can_bundle_canned_repository(self, mock_default_catalog, mock_catalogs):
+        mock_catalog_name = 'testcanned'
+        mock_catalog = config.Catalog(name=mock_catalog_name,
+                                      atlas='hca',
+                                      plugins={
+                                          'metadata': 'hca',
+                                          'repository': 'canned'
+                                      })
+        mock_default_catalog.return_value = mock_catalog_name
+        mock_catalogs.return_value = {mock_catalog_name: mock_catalog}
+        self._test_catalog(mock_catalog)
+
+    def bundle_fqid(self, catalog: CatalogName) -> SourcedBundleFQID:
+        plugin = RepositoryPlugin.load(catalog).create(catalog)
+        spec = first(plugin.sources)
+        source = plugin.resolve_source(str(spec))
+        return first(plugin.list_bundles(source, prefix=source.spec.prefix))
```

Comment:
This probably isn't worth troubling about, but the list_bundles query fetches *all* the bundles with that prefix when you're only trying to get one. Ideally the size could be specified in the query.

Reply:
> list_bundles query fetches all the bundles with that prefix

This might be just a confusion of terminology, but `RepositoryPlugin.list_bundles` does not fetch any bundles, it only lists the bundle FQIDs. `RepositoryPlugin.fetch_bundle` is a separate method.

Reply:
It still lists all bundles even though it only needs one. This will definitely cause trouble in dev and prod where no prefix is used on the IT catalogs and 100k+ bundles will be returned.  The indexing IT uses a technique to shrink a randomly chosen prefix until 64 bundles match. You should use that or something like that here too. Second, it would be nice to use a randomly selected bundle but the indexing test prefix selection already takes care of that. As always, it should be possible to seed the random generator with minimal code changes in order to be able to reproduce IT failures.